sdk/src/assertions/soft_binding.rs

```diff
@@ -0,0 +1,167 @@
+use serde::{Deserialize, Serialize};
+
+use super::labels;
+use crate::{
+    assertion::{Assertion, AssertionBase, AssertionCbor},
+    assertions::region_of_interest::RegionOfInterest,
+    cbor_types::UriT,
+    Result,
+};
+
+const ASSERTION_CREATION_VERSION: usize = 1;
+
+/// The data structure used to store one or more soft bindings across some or all of the asset's content.
+///
+/// https://c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html#soft_binding_assertion
+#[derive(Serialize, Deserialize, Debug, Default, PartialEq)]
+pub struct SoftBinding {
+    /// A string identifying the soft binding algorithm and version of that algorithm used to compute the value,
+    /// taken from the [C2PA soft binding algorithm list](https://github.com/c2pa-org/softbinding-algorithm-list).
+    ///
+    /// If this field is absent, the algorithm is taken from the `alg_soft` value of the enclosing structure.
+    /// If both are present, the field in this structure is used. If no value is present in any of these places,
+    /// this structure is invalid; there is no default.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub alg: Option<String>,
+
+    /// A list of details about the soft binding.
+    pub blocks: Vec<SoftBindingBlockMap>,
+
+    /// A human-readable description of what this hash covers.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+
+    /// A string describing parameters of the soft binding algorithm.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub alg_params: Option<String>,
+
+    /// Zero-filled bytes used for filling up space.
+    #[serde(with = "serde_bytes")]
+    pub pad: Vec<u8>,
+
+    /// Zero-filled bytes used for filling up space.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pad2: Option<serde_bytes::ByteBuf>,
+
+    #[serde(skip_serializing)]
+    url: Option<UriT>,
+}
+
+impl SoftBinding {
+    /// A file or http(s) URL to where the bytes that are being hashed lived.
+    ///
+    /// This is useful for cases where the data lives in a different file chunk or side-car
+    /// than the claim.
+    #[deprecated]
+    pub fn url(&self) -> Option<&UriT> {
```

Comment:
It seems very bizarre to have a deprecated function on a brand new assertion. What is the logic here?
Also when using the deprecated macro, we should specify when it was deprecated and what to use instead.

Reply:
The spec deprecated this in 1.3 and says to use the asset reference assertion instead. I marked it as deprecated to signify to users that it shouldn't be used. We do support reading older manifests which is why this field is here although it's not recommended hence the deprecation.